# SDLFM_ReverseEngineering

The toolbox contains the implementation of the switched latent force model for reverse-engineering transcriptional regulation in gene expression data proposed in [1]. This approach is inspired by the framework from [2,3], using the mechanistic model from [4].

### Dependences (toolboxes):
1. Switchedlfm toolbox: this toolbox is a modified version from the one available in the GPmat toolbox.
1. Some dependences from GPmat toolbox (https://github.com/SheffieldML/GPmat): gp, kern, mltools, multigp2, ndlutil, optimi.
2. Netlab toolbox (https://github.com/sods/netlab).

### References:

[1] A. F. López-Lopera, and M. A. Álvarez (2017). Switched Dynamical Latent Force Models for Modelling Transcriptional Regulation. Computational Biology and Bioinformatics, IEEE/ACM Transactions on (to appear).

[2] M. A. Álvarez, D. Luengo, and N. D. Lawrence (2013). Linear Latent Force Models Using Gaussian Processes. Pattern Analysis and Machine Intelligence, IEEE Transactions on, 35(11): 2693–2705.

[3] M. A. Álvarez, J. R. Peters, N. D. Lawrence, and B. Schölkopf (2010). Switched Latent Force Models for Movement Segmentation. In Advances in Neural Information Processing Systems 23, edited by J.D. Lafferty, C.K.I. Williams, J. Shawe-Taylor, R.S. Zemel, and A. Culotta, 55–63. Curran Associates, Inc.

[4] N. D. Lawrence, G. Sanguinetti, and M. Rattray (2006). Modelling Transcriptional Regulation Using Gaussian Processes. In Advances in Neural Information Processing Systems 19, edited by B. Schölkopf, J. C. Platt, and T. Hoffman, 785–92. MIT Press.
